code/game/machinery/_machines_base/stock_parts/network_lock.dm

```diff
@@ -83,15 +98,48 @@
 		return
 	data["connected"] = TRUE
 	data["default_state"] = auto_deny_all
-	var/list/grants_data = list()
 	if(!network.access_controller)
 		return
-	for(var/datum/computer_file/data/grant_record/GR in network.access_controller.get_all_grants())
-		grants_data.Add(list(list(
-			"grant_name" = GR.stored_data,
-			"assigned" = (GR.stored_data in grants)
-		)))
-	data["grants"] = grants_data
+
+	data["patterns"] = list()
+	var/pattern_index = 0
+	for(var/list/pattern in groups)
+		pattern_index++
+		data["patterns"].Add(list(list(
+			"index" = "[pattern_index]",
+			"groups" = english_list(pattern, "No groups assigned!") 
```

Comment:
should pass and_text="or" here, otherwise result is misleading in UI (groups are OR based but it lists them as AND in text)